src/mongo/util/lru_cache.h

```diff
@@ -131,7 +131,7 @@ class LRUCache {
      */
     const_iterator cfind(const K& key) const {
         auto it = this->_map.find(key);
-        return (it == this->_map.end()) ? this->end() : it->second;
+        return (it == this->_map.end()) ? this->end() : const_iterator(it->second);
```

Comment:
I'd like to add a comment above here:

// TODO(SERVER-28890): Remove the function-style cast when MSVC's
//`std::list< ... >::iterator` implementation doesn't conflict with their `/Zc:ternary`